File: 45주차/기능개발.js
function solution(progresses, speeds) {
  var answer = [];
  var q = [];
  for (let i = 0; i < progresses.length; i++) {
    q.push(Math.ceil((100 - progresses[i]) / speeds[i])); // 남을 일수 구하기.
  }
  console.log(q);
  while (q.length > 0) {
    var s = [];
    var target = q.shift();
    s.push(target);
    while (q && target >= q[0]) {
      //한번 배포될 때, 여러개 배포체크를 위해 s에 push.
      s.push(q.shift());
    }
    answer.push(s.length);
  }
  return answer;
}
// 프로그래머스 js로 풀기.

File: 66주차/기능개발.py
import math
def solution(progresses,speeds):
    answer = []
    n = len(speeds)
    arr = []
    stack = []
    for i in range(n):
        v = (100- progresses[i])/speeds[i]
        arr.append(math.ceil(v))
    cnt = 1
    for i in range(len(arr)):
        if len(stack) == 0:
            stack.append(arr[i])
        else:
            if stack[-1] >= arr[i]:# 앞의 기능보다 같거나 적게 걸리면 함께 배포되므로 cnt+1
                cnt += 1
            else:# 더 걸린다면, pop해서 answer에 개수 push
                stack.pop()
                stack.append(arr[i])
                answer.append(cnt)
                cnt = 1
    answer.append(cnt)
    return answer